packages/js/uri-resolvers/src/helpers/ResolverWithLoopGuard.ts

```diff
@@ -0,0 +1,46 @@
+import { InfiniteLoopError } from "./InfiniteLoopError";
+import { UriResolverLike } from "../helpers";
+import { buildUriResolver } from "../utils";
+
+import { Result } from "@polywrap/result";
+import {
+  IUriResolver,
+  Uri,
+  Client,
+  IUriResolutionContext,
+  UriPackageOrWrapper,
+  UriResolutionResult,
+} from "@polywrap/core-js";
+
+export class ResolverWithLoopGuard<TError = undefined>
```

Comment:
I think this file is not being used but I might be wrong :thinking:

Reply:
It's not, but it's a useful primitive for users :)